### PR TITLE
[ggphp] feat(mixins): Add mixins to the PHP microgenerator

### PR DESCRIPTION
**Pull request by miraleung**
Reference for generated diffs, compared to a non-mixin RPC:
- [Client changes](https://github.com/googleapis/google-cloud-php-pubsub/blob/master/src/V1/Gapic/PublisherGapicClient.php#L1076)
- [Test changes](https://github.com/googleapis/google-cloud-php-pubsub/blob/master/tests/Unit/V1/PublisherClientTest.php#L816)

### CODE REVIEW

File: src/CodeGenerator.php [hunk 1]
```diff
@@ -38,6 +38,11 @@
 
 class CodeGenerator
 {
+    const MIXIN_SERVICES =  [
+        "google.iam.v1.IAMPolicy",
+          "google.longrunning.Operations",
+          "google.cloud.location.Locations"
```
Comment (noahdietz) [inline src/CodeGenerator.php:44]: Nit: consistent indentation

File: src/Generation/GapicClientGenerator.php [hunk 2]
```diff
@@ -74,6 +74,8 @@ private function generateImpl(): PhpFile
             }
         }
         foreach ($this->serviceDetails->methods as $method) {
+            if ($method->requestType === null) {
+            }
```
Comment (noahdietz) [inline src/Generation/GapicClientGenerator.php:78]: Nit: empty statement?

File: tests/Integration/apis/kms/v1/cloudkms_test_mixins_v1.yaml [hunk 3]
```diff
@@ -0,0 +1,75 @@
+type: google.api.Service
+config_version: 3
+name: cloudkms.googleapis.com
+title: Cloud Key Management Service (KMS) API
+
+apis:
+- name: google.cloud.kms.v1.KeyManagementService
```
Comment (noahdietz) [inline tests/Integration/apis/kms/v1/cloudkms_test_mixins_v1.yaml:7]: Should this have the google.iam.v1.IAMPolicy mixin?

File: tests/Integration/goldens/kms/src/V1/Gapic/KeyManagementServiceGapicClient.php [hunk 4]
```diff
@@ -2285,7 +2286,7 @@ public function getLocation(array $optionalArgs = [])
             'name' => $request->getName(),
         ]);
         $optionalArgs['headers'] = isset($optionalArgs['headers']) ? array_merge($requestParams->getHeader(), $optionalArgs['headers']) : $requestParams->getHeader();
-        return $this->startCall('GetLocation', Location::class, $optionalArgs, $request)->wait();
+        return $this->startCall('GetLocation', Location::class, $optionalArgs, $request, Call::UNARY_CALL, 'google.cloud.location.Locations')->wait();
```
Comment (noahdietz) [inline tests/Integration/goldens/kms/src/V1/Gapic/KeyManagementServiceGapicClient.php:2289]: Where is the Location stuff coming from when the mixin is not included in the service yaml?

**Review by noahdietz — COMMENTED** [review]
Some nits and questions

File: src/Generation/GapicClientGenerator.php (hunk 2 above)
Comment (miraleung) [reply src/Generation/GapicClientGenerator.php]: Done.

File: tests/Integration/apis/kms/v1/cloudkms_test_mixins_v1.yaml (hunk 3 above)
Comment (miraleung) [reply tests/Integration/apis/kms/v1/cloudkms_test_mixins_v1.yaml:7]: Fixed. Keeping Locations in and leaving IAM out, to ensure that (a) this list is actually parsed, and (b) Locations has both regular and paginated RPCs.

File: tests/Integration/goldens/kms/src/V1/Gapic/KeyManagementServiceGapicClient.php (hunk 4 above)
Comment (miraleung) [reply tests/Integration/goldens/kms/src/V1/Gapic/KeyManagementServiceGapicClient.php:2129]: Fixed.

**Review by miraleung — COMMENTED** [review]
Please ignore the `gapic_metadata.json` files, they should be resolved once parent branches have been merged (as local git merges seemed to have no effect).

File: tests/Integration/BUILD.bazel [hunk 5]
```diff
@@ -238,21 +222,53 @@ php_gapic_library(
...
     deps = [
-        ":iam_php_grpc",
+        #":iam_php_grpc",
         ":kms_php_gapic",
         ":kms_php_grpc",
         ":kms_php_proto",
-        ":kms_proto",
-        "@com_google_googleapis//google/cloud/location:location_php_grpc",
-        "@com_google_googleapis//google/cloud/location:location_php_proto",
+        #"@com_google_googleapis//google/cloud/location:location_php_grpc",
```
Comment (noahdietz) [inline tests/Integration/BUILD.bazel:236]: Can you include a comment as to why these are commented out? Like what was done later on in the service yaml.

**Review by noahdietz — APPROVED** [review]

**Pull request merged**